### ISSUE TITLE
New Page: Extended splash screen

### ISSUE THREAD

**Issue by mrlacey** [Microsoft/WindowsTemplateStudio#293]
Add a new page displayed when the app first launches to act as an animated/extended splash screen.
Such a requirement is common for apps that need to connect to a backend server before starting or perform other actions before they can be used.

**Comment by mrlacey** [comment]
I have a PoC of this and have shown it to Clint & Michael. Will add as a branch next week when I get back from build.

**Comment by jamesmcroft** [comment]
I'll be looking for the PR to take a look through. I use my own implementation of extended splash in apps and will definitely have feedback :smile:

**Comment by mrlacey** [comment]
For reference this needs #294 completing first

**Comment by mrlacey** [comment]
Removing myself as assignee as don't want to block others from moving this forward.

**Comment by mrlacey** [comment]
Also, for reference see https://docs.microsoft.com/en-us/windows/uwp/launch-resume/create-a-customized-splash-screen

**Comment by Paradoxon101** [comment]
I think this is almost necessary if you also want to run on the XBOX. There UWP apps start much slower and are then stopped by the system if the app takes too long to start.

Is there any news as to whether the page will be added?

**Comment by mrlacey** [comment]
@Paradoxon101 no-one had expressed an interest in this and so it has been languishing on the backlog.

Are you saying that apps created with WinTS do not launch on the Xbox? Have you got an example of the options you used to create an app that had this issue? 
If this is occurring, it's something we will look to address separately as there should be nothing in a generated app that prevents the app from launching on any device.

I don't think the need to have this feature is directly tied to Xbox, it's tied to apps that can only work online or must do a lot of work on startup. General guidance for any app is to allow working in some form even if offline (although this can be limiting for some games or collaborative apps) and to minimize the work that is done at start-up, so to avoid any launch delays.

If you have specific scenarios or needs that you can share it will help us prioritize this to be done sooner.


---

_note to self: this will probably be further impacted by future identity/authentication work._

**Comment by Paradoxon101** [comment]
> I don't think the need to have this feature is directly tied to Xbox, it's tied to apps that can only work online or must do a lot of work on startup.

Yeah, that's what I meant.

**Comment by Moreno-Borsalino** [comment]
I need an extended splash page in a UWP app but I don't understand how to modify the code to insert this page. It's a shame that there isn't an option to create it in the template.
Could someone explain to me what to change in App.xaml.cs to add the extended splash screen page to load data when the app starts? Thanks to anyone who wants to give me some suggestions
